api_types.ts

```diff
@@ -546,6 +547,35 @@ export interface CommonPackFormulaDef<T extends ParamDefs> {
    * @hidden
    */
   readonly allowedAuthenticationNames?: string[];
+
+  /**
+   * The JavaScript function that implements parameter validation.
+   *
+   * This function takes in parameters and a context containing a {@link PermissionSyncMode}
+   * and validates the parameters. A formula may want to validate parameters differently
+   * for permissionSyncMode 'PermissionAware' vs 'Personal' vs undefined (which represents a formula).
+   *
+   * @returns if the parameters are valid
+   * @throws {@link ParameterValidationError} if the parameters are invalid.
+   *
+   * @example
+   * ```
+   * valiateParameters: async function (context, _, formulaContext) {
+   *   let [quantity, sku] = formulaContext?.params;
+   *   let errors = [];
+   *   if (quantity < 0) {
+   *     errors.push({message: "Must be a positive number.", propertyName: "quantity"});
+   *   }
+   *   if (!isValidSku(context, sku)) {
+   *     errors.push({message: `Product SKU not found.`, propertyName: "sku"});
+   *   }
+   *   if (errors.length > 0) {
+   *     throw new ParameterValidationError("Invalid parameter values.", errors);
+   *   }
+   * },
+   * ```
+   */
+  validateParameters?: MetadataFunction<ExecutionContext, boolean>;
```

Comment:
Hmm, if we want to keep this consistent with other metadata formulas, wouldn't this be `MetadataFormulaDef`, and then we'd auto-wrap this with `makeMetadataFormula` if we see the function version like we do for other metadata formulas?

Right now it looks like you're wrapping only when making the pack metadata, but for other metadata formulas we wrap and convert to a formula object at import time. I agree that the object representation is kind of obsolete at this point but I think we might as well be consistent.

Reply:
`wrapMetadataFunction` seems to be what I needed, didn't know that existed before